src/createEslinter.ts

```diff
@@ -5,11 +5,11 @@ import { throwIfIsInvalidSourceFileError } from './FsHelper';
 import { makeCreateNormalizedMessageFromEsLintFailure } from './NormalizedMessageFactories';
 import * as path from 'path';
 
-export function createEslinter(eslintOptions: object) {
+export function createEslinter(eslintOptions: eslinttypes.CLIEngine.Options) {
   // tslint:disable-next-line:no-implicit-dependencies
   const eslint: typeof eslinttypes = require('eslint');
```

Comment:
@johnnyreilly you conditionally require `eslint` here, your argument that you then need to make `ESLint` a dependency isn't valid as it is not a top-level import and eslint above is used as the type.